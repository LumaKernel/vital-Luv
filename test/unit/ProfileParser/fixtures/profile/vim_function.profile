SCRIPT  /test/test_function.vim
Sourced 1 time
Total time:   0.000149
 Self time:   0.000142

count  total (s)   self (s)
                            
    1              0.000003 function! F()
                              echomsg 'string'
                            endfunction
                            
    1   0.000050   0.000043 call F()
                            

FUNCTION  F()
    Defined: /test/original/some/../../test_function.vim:2
Called 1 time
Total time:   0.000008
 Self time:   0.000008

count  total (s)   self (s)
    1              0.000005   echomsg 'string'

FUNCTIONS SORTED ON TOTAL TIME
count  total (s)   self (s)  function
    1   0.000008             F()

FUNCTIONS SORTED ON SELF TIME
count  total (s)   self (s)  function
    1              0.000008  F()

