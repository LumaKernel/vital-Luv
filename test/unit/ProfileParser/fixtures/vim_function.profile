SCRIPT  /home/luma/vim-make-plugin/vital-Luv/dev/ProfileParser/test_function.vim
Sourced 1 time
Total time:   0.000172
 Self time:   0.000150

count  total (s)   self (s)
                            
    1              0.000022 function! F()
                              echomsg 'string'
                            endfunction
                            
    1   0.000041   0.000019 call F()
                            

FUNCTION  F()
    Defined: ~/vim-make-plugin/vital-Luv/dev/ProfileParser/test_function.vim:2
Called 1 time
Total time:   0.000022
 Self time:   0.000022

count  total (s)   self (s)
    1              0.000019   echomsg 'string'

FUNCTIONS SORTED ON TOTAL TIME
count  total (s)   self (s)  function
    1   0.000022             F()

FUNCTIONS SORTED ON SELF TIME
count  total (s)   self (s)  function
    1              0.000022  F()

