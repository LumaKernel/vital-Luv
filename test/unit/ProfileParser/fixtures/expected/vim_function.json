[{"path":"/test/test_function.vim","total_time":1.49e-4,"self_time":1.42e-4,"type":"script","lines":[{"content":""},{"self_time":3.0e-6,"count":1,"content":"function! F()"},{"content":"  echomsg 'string'"},{"content":"endfunction"},{"content":""},{"total_time":5.0e-5,"self_time":4.3e-5,"count":1,"content":"call F()"},{"content":""}],"count":1},{"defined":{"path":"/test/test_function.vim","line":2},"name":"F()","total_time":8.0e-6,"self_time":8.0e-6,"type":"function","lines":[{"self_time":5.0e-6,"count":1,"content":"  echomsg 'string'"}],"count":0},{"type":"function_list","what":"TOTAL TIME","functions":[{"name":"F()","total_time":8.0e-6,"count":1}]},{"type":"function_list","what":"SELF TIME","functions":[{"name":"F()","self_time":8.0e-6,"count":1}]}]
